specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,267 @@
+  COREWEBVIEW2_SAVE_AS_INVALID_PATH,
+  /// Save as downloading not start as given a duplicate filename and
+  /// replace file not allowed
+  COREWEBVIEW2_SAVE_AS_FILE_ALREADY_EXISTS,
+  /// Save as downloading not start as the `SAVE_AS_TYPE` selection not
+  /// supported because of the content MIME type or system limits
+  COREWEBVIEW2_SAVE_AS_TYPE_NOT_SUPPORTED,
+  /// Cancel the save as request
+  COREWEBVIEW2_SAVE_AS_CANCELLED,
+  /// Save as request complete, the downloading started
+  COREWEBVIEW2_SAVE_AS_STARTED
+} COREWEBVIEW2_SAVE_CONTENT_AS_RESULTS;
+
+
+[uuid(15e1c6a3-c72a-4df3-91d7-d097fbec3bfd), object, pointer_default(unique)]
+interface ICoreWebView2Staging20 : IUnknown {
+  /// Programmatically trigger a save as action for current content.
+  ///
+  /// Opens a system modal dialog by default. Returns COREWEBVIEW2_SAVE_AS_OPEN_SYSTEM_DIALOG.
+  /// If it was already opened, this method would not open another one
+  ///
+  /// If the silent save as option is enabled, won't open the system dialog, will
+  /// raise the `SaveAsRequested` event instead and process through its event args. The method can return
+  /// a detailed info to indicate the call's result. Please see COREWEBVIEW2_SAVE_CONTENT_AS_RESULTS
+  ///
+  /// \snippet ScenarioSaveAs.cpp ProgrammaticSaveAs
+  HRESULT SaveContentAs([in] ICoreWebView2StagingSaveContentAsCompletedHandler* handler);
+
+  /// Add an event handler for the `SaveAsRequested` event. This event is raised
+  /// when save as is triggered, programmatically or manually.
+  ///
+  /// \snippet ScenarioSaveAs.cpp ToggleSilent
+  HRESULT add_SaveAsRequested(
+   [in] ICoreWebView2StagingSaveAsRequestedEventHandler* eventHanlder,
+   [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_SaveAsRequested`.
+  ///
+  /// \snippet ScenarioSaveAs.cpp ToggleSilent
+  HRESULT remove_SaveAsRequested(
+   [in] EventRegistrationToken token);
+}
+
+/// The event handler for the `SaveAsRequested` event, when the handler
+/// exists, the silent save as enables.
+[uuid(55b86cd2-adfd-47f1-9cef-cdfb8c414ed3), object, pointer_default(unique)]
+interface ICoreWebView2StagingSaveAsRequestedEventHandler : IUnknown {
+  HRESULT Invoke(
+   [in] ICoreWebView2* sender,
+   [in] ICoreWebView2StagingSaveAsRequestedEventArgs* args);
+}
+
+/// The event args for `SaveAsRequested` event
+[uuid(80101027-b8c3-49a1-a052-9ea4bd63ba47), object, pointer_default(unique)]
+interface ICoreWebView2StagingSaveAsRequestedEventArgs : IUnknown {
+  /// Get the Mime type of content to be saved
+  [propget] HRESULT ContentMimeType([out, retval] LPWSTR* value);
+
+  /// Indicates if this event is a silent save as job. `Handled` as FALSE means
+  /// save as handled by system default dialog; TRUE means a silent save as,
+  /// will skip the system dialog.
+  ///
+  /// Set the `Handled` for save as
+  [propput] HRESULT Handled ([in] BOOL handled);
```

Comment:
We have overloaded `Handled` previously in our events to mean various things. It would be better to use an explicit name here to clearly indicate what it does. In this case how about

```suggestion
  [propput] HRESULT SupressDefaultDialog([in] BOOL value);
```

Reply:
I'm confused. The description of this Handled property makes it sound like, if you set Handled to true there is no default dialog allowing the end developer to show their own custom UI. That is great behavior, we just want to be more explicit and name it something like SupressDefaultDialog. Are you saying this property doesn't do that?